src/components/feedback/Dialog.tsx

```diff
@@ -0,0 +1,158 @@
+import classnames from 'classnames';
+import type { RefObject } from 'preact';
+import { useEffect, useLayoutEffect } from 'preact/hooks';
+
+import { useElementShouldClose } from '../../hooks/use-element-should-close';
+import { useSyncedRef } from '../../hooks/use-synced-ref';
+import { useUniqueId } from '../../hooks/use-unique-id';
+import type { PresentationalProps } from '../../types';
+import { downcastRef } from '../../util/typing';
+import Overlay from '../layout/Overlay';
+import Panel from '../layout/Panel';
+import type { PanelProps } from '../layout/Panel';
+
+type ComponentProps = {
+  /**
+   * Dialog _should_ be provided with a close handler. We have a few edge use
+   * cases, however, in which we need to render a "non-closeable" modal.
```

Comment:
```suggestion
   * cases, however, in which we need to render a "non-closeable" dialog.
```

Reply:
Accccctually, this is probably technically correct (though its inclusion was an accident). Only modal dialogs can be non-close-able :)